build/native.rs

```diff
@@ -33,35 +31,76 @@ const DEFAULT_SDK_DIR: &str = ".espressif";
 const DEFAULT_ESP_IDF_REPOSITORY: &str = "https://github.com/espressif/esp-idf.git";
 const DEFAULT_ESP_IDF_VERSION: &str = "v4.3";
 
-fn esp_idf_version() -> git::Ref {
-    let version =
-        env::var(ESP_IDF_VERSION_VAR).unwrap_or_else(|_| DEFAULT_ESP_IDF_VERSION.to_owned());
-    let version = version.trim();
-    assert!(
-        !version.is_empty(),
-        "${} (='{}') must contain a valid version",
-        ESP_IDF_VERSION_VAR,
-        version
-    );
+const CARGO_CMAKE_BUILD_ACTIVE_VAR: &str = "CARGO_CMAKE_BUILD_ACTIVE";
+const CARGO_CMAKE_BUILD_INCLUDES_VAR: &str = "CARGO_CMAKE_BUILD_INCLUDES";
+const CARGO_CMAKE_BUILD_LINK_LIBRARIES_VAR: &str = "CARGO_CMAKE_BUILD_LINK_LIBRARIES";
+const CARGO_CMAKE_BUILD_COMPILER_VAR: &str = "CARGO_CMAKE_BUILD_COMPILER";
+const CARGO_CMAKE_BUILD_SDKCONFIG_VAR: &str = "CARGO_CMAKE_BUILD_SDKCONFIG";
 
-    match version.split_once(':') {
-        Some(("commit", c)) => git::Ref::Commit(c.to_owned()),
-        Some(("tag", t)) => git::Ref::Tag(t.to_owned()),
-        Some(("branch", b)) => git::Ref::Branch(b.to_owned()),
-        _ => match version.chars().next() {
-            Some(c) if c.is_ascii_digit() => git::Ref::Tag("v".to_owned() + version),
-            Some('v') if version.len() > 1 && version.chars().nth(1).unwrap().is_ascii_digit() => {
-                git::Ref::Tag(version.to_owned())
-            }
-            Some(_) => git::Ref::Branch(version.to_owned()),
-            _ => unreachable!(),
-        },
+pub fn build() -> Result<EspIdfBuildOutput> {
+    if env::var(CARGO_CMAKE_BUILD_ACTIVE_VAR).is_ok()
+        || env::var(CARGO_CMAKE_BUILD_INCLUDES_VAR).is_ok()
+    {
+        build_cmake_first()
+    } else {
+        build_cargo_first()
     }
 }
 
-pub(crate) fn main() -> Result<EspIdfBuildOutput<impl Iterator<Item = (String, kconfig::Value)>>> {
-    // TODO: Implement support for CMake-first builds in a similar way as to how cargo-pio supports PIO-first builds
+fn build_cmake_first() -> Result<EspIdfBuildOutput> {
+    const IDF_LIB_PREFIX: &str = "__idf_";
+
+    let enabled_components = EspIdfComponents::from(
+        env::var(CARGO_CMAKE_BUILD_LINK_LIBRARIES_VAR)?
+            .split(";")
+            .filter_map(|s| {
+                if s.starts_with(IDF_LIB_PREFIX) {
```

Comment:
I think you want [`str::strip_prefix`](https://doc.rust-lang.org/std/primitive.str.html#method.strip_prefix) here.

Reply:
Indeed - will fix.